state/keys.go

```diff
@@ -75,21 +80,48 @@ func (k *Keys) UnmarshalJSON(b []byte) error {
 	if err := json.Unmarshal(b, &keysJSON); err != nil {
 		return err
 	}
-	for perm, keyList := range keysJSON.Perms {
-		if perm < int(None) || perm > int(All) {
+	for strPerm, keyList := range keysJSON.Perms {
+		perm, err := newPermissionFromString(strPerm)
+		if err != nil {
+			return err
+		}
+		if perm < None || perm > All {
 			return fmt.Errorf("invalid permission encoded in json %d", perm)
 		}
 		for _, encodedKey := range keyList {
 			key, err := hex.DecodeString(encodedKey)
 			if err != nil {
 				return err
 			}
-			(*k)[string(key)] = Permissions(perm)
+			(*k)[string(key)] = perm
 		}
 	}
 	return nil
 }
 
+func newPermissionFromString(s string) (Permissions, error) {
+	var perm Permissions
+	switch s {
+	case "read":
+		perm = Read
+	case "write":
+		perm = Write
+	case "allocate":
+		perm = Allocate
+	case "all":
+		perm = All
+	case "none":
+		perm = None
+	default:
+		res, err := hex.DecodeString(s)
+		if err != nil || len(res) != 1 {
+			return 0, fmt.Errorf("permission %s: %w", s, errInvalidHexadecimalString)
+		}
+		perm = Permissions(res[0])
+	}
+	return perm, nil
```

Comment:
Rather than adding this function, could we implement the `TextMarshaler` and `TextUnmarshaler` interfaces on `Permissions` so that we can use `Permissions` in a type we're going to encode to JSON and it will produce the human readable format? See the second example here: https://pkg.go.dev/encoding/json